src/value.cpp

```diff
@@ -209,6 +209,19 @@ Value::Value(const DataType::ConstPtr& data_type, Decoder decoder)
   }
 }
 
+bool Value::update(const Decoder& decoder) {
+  decoder_ = decoder;
+  if (!decoder_.is_null()) {
+    if (data_type_->is_collection()) {
```

Comment:
This also needs to handle tuple and UDT cases.

Reply:
Ahh indeed as it resets the count on a null value

Reply:
Fixed, I think there is an intermittent unittest failure

Reply:
@mpenick could you please review the fix

Reply:
I'm reviewing and running through a couple integration tests.